server/app/controllers/fusor/api/customer_portal/customer_portal_proxies_controller.rb

```diff
@@ -32,6 +32,15 @@ def logout
           render :json => {}
         end
 
+        def is_authenticated
+          authenticated = false
+          if session[:portal_username] && session[:portal_password]
+            # they are not nil, let's make sure neither is empty
+            authenticated = session[:portal_username].empty? && session[:portal_password].empty?
```

Comment:
I'm a little confused.  The code looks like "authenticated is true if both are empty", but the comment above says "make sure neither is empty".